server/NucleiSegmentation/NucleiSegmentation.py

```diff
@@ -17,15 +18,33 @@
 }
 
 
+def read_image(file):
+    return skimage.io.imread(file)[:, :, :3]
+
+
 def main(args):
 
     #
     # Read Input Image
     #
     print('>> Reading input image')
 
-    imInput = skimage.io.imread(args.inputImageFile)[:, :, :3]
-
+    imInput = None
+    imageFile = args.inputImageFile
+    if os.path.isdir(imageFile):
+        dir = imageFile
+        for file in os.listdir(dir):
```

Comment:
Items that are tagged by the large_image plugin contain a `largeImage.fileId` property that we should use.  The reason I couldn't use it for the purposes of this demo was because the analysis image couldn't handle the tiled image file produced by the large image plugin.